parity/cli/mod.rs

```diff
@@ -934,198 +810,109 @@ usage! {
-
-			FLAG flag_ipc_off: (bool) = false, or |_| None,
-			"--ipc-off",
-			"Equivalent to --no-ipc.",
-
 			FLAG flag_testnet: (bool) = false, or |_| None,
 			"--testnet",
 			"Testnet mode. Equivalent to --chain testnet. Overrides the --keys-path option.",
 
-			FLAG flag_nodiscover: (bool) = false, or |_| None,
-			"--nodiscover",
-			"Equivalent to --no-discovery.",
-
-			// FLAG Removed in 1.7.
-
-			FLAG flag_dapps_apis_all: (bool) = false, or |_| None,
-			"--dapps-apis-all",
-			"Dapps server is merged with HTTP JSON-RPC server. Use --jsonrpc-apis.",
-
-			// FLAG Removed in 1.11.
-
-			FLAG flag_public_node: (bool) = false, or |_| None,
-			"--public-node",
-			"Does nothing; Public node is removed from Parity.",
-
-			FLAG flag_force_ui: (bool) = false, or |_| None,
-			"--force-ui",
-			"Does nothing; UI is now a separate project.",
-
-			FLAG flag_no_ui: (bool) = false, or |_| None,
-			"--no-ui",
-			"Does nothing; UI is now a separate project.",
-
-			FLAG flag_ui_no_validation: (bool) = false, or |_| None,
-			"--ui-no-validation",
-			"Does nothing; UI is now a separate project.",
-
-			// FLAG Removed in 2.0.
-
-			FLAG flag_fast_and_loose: (bool) = false, or |_| None,
-			"--fast-and-loose",
-			"Does nothing; DB WAL is always activated.",
-
-			FLAG flag_no_dapps: (bool) = false, or |c: &Config| c.dapps.as_ref()?._legacy_disable.clone(),
-			"--no-dapps",
-			"Disable the Dapps server (e.g. status page).",
-
-			// ARG Removed in 1.6 or before.
-
-			ARG arg_etherbase: (Option<String>) = None, or |_| None,
-			"--etherbase=[ADDRESS]",
-			"Equivalent to --author ADDRESS.",
-
-			ARG arg_extradata: (Option<String>) = None, or |_| None,
-			"--extradata=[STRING]",
-			"Equivalent to --extra-data STRING.",
-
-			ARG arg_datadir: (Option<String>) = None, or |_| None,
-			"--datadir=[PATH]",
-			"Equivalent to --base-path PATH.",
-
-			ARG arg_networkid: (Option<u64>) = None, or |_| None,
-			"--networkid=[INDEX]",
-			"Equivalent to --network-id INDEX.",
-
-			ARG arg_peers: (Option<u16>) = None, or |_| None,
-			"--peers=[NUM]",
-			"Equivalent to --min-peers NUM.",
-
-			ARG arg_nodekey: (Option<String>) = None, or |_| None,
-			"--nodekey=[KEY]",
-			"Equivalent to --node-key KEY.",
-
-			ARG arg_rpcaddr: (Option<String>) = None, or |_| None,
-			"--rpcaddr=[IP]",
-			"Equivalent to --jsonrpc-interface IP.",
+			// Light client functionality was deprecated in 3.0.
+			FLAG flag_light: (bool) = false, or |c: &Config| c.parity.as_ref()?.light,
+			"--light",
+			"Experimental: run in light client mode. Light clients synchronize a bare minimum of data and fetch necessary data on-demand from the network. Much lower in storage, potentially higher in bandwidth. Has no effect with subcommands.",
 
-			ARG arg_rpcport: (Option<u16>) = None, or |_| None,
-			"--rpcport=[PORT]",
-			"Equivalent to --jsonrpc-port PORT.",
+			FLAG flag_no_hardcoded_sync: (bool) = false, or |c: &Config| c.parity.as_ref()?.no_hardcoded_sync,
```

Comment:
This is light client-related also, should probably have the same comment as `--light`

Reply:
Isn't there a sub command that is used to produce the hard coded headers too? Needs deprecating too then?

Reply:
Do you mean `export-hardcoded-sync`? Then yes